run-command.h

```diff
@@ -49,6 +49,7 @@ struct child_process {
 	unsigned use_shell:1;
 	unsigned clean_on_exit:1;
 	unsigned wait_after_clean:1;
+	unsigned close_parent_file_descriptors:1;
```

Comment:
I don't know about that name... I read it initially as *all* of the parent's file descriptors, but that is not true. For one, the fds that need to be inherited (because they are connected to a pipe so that the spawning process can talk to the spawned process) need to not be closed. And then, we only close file descriptors referring to files. If there is a socket open, the code typically still talks about "file" descriptors, but we do not close those.

Maybe `restrict_fd_inheritance`. Nah. Also not very good. I am not very good at naming things.

Reply:
I can't think of anything better than 'close_parent_file_descriptors' - it doesn't say "all" file descriptors so someone will have to look at the code/commit/comments if they want to know which ones. :)